homeassistant/components/prometheus.py

```diff
@@ -212,8 +221,15 @@ def _handle_switch(self, state):
             self.prometheus_client.Gauge,
             'State of the switch (0/1)',
         )
-        value = state_helper.state_as_number(state)
-        metric.labels(**self._labels(state)).set(value)
+
+        if state.state == STATE_UNAVAILABLE:
```

Comment:
While testing a new client version I noticed that state_to_number doesn't handle STATE_UNAVAILABLE and rather than changing a core helper that could cause more damage that good I opted for this quick fix to cut down on the exception thrown.

Reply:
Why only add this for switches? It could happen for any entity type.

Reply:
I only put it on switches since I can't assume that STATE_UNAVAILABLE means the same for other domains. A switch is either on or off so it's easier to make this assumption. I could do the same for binary_sensors though

Reply:
`STATE_UNAVAILABLE` is set by the base entity class. It always means the same thing.